feat: add option to change base url of github

BlackDark opened the pull request:
Added new parameter for specifying a different URL for github.
This is useful if a github enterprise instance is hosted and you want to use the code there.

No breaking change.

gagoar commented:
Hey, I just saw that, would you like to finish this? I see there are some problems with the workflows, I can fix that!

gagoar commented:
Hey @BlackDark I've fixed some of the workflows, to make them run correctly when there's a fork, could you rebase with main? thanks!

BlackDark commented:
> Hey @BlackDark I've fixed some of the workflows, to make them run correctly when there's a fork, could you rebase with main? thanks!
> 
> Also please add a description of the use case you are trying to get this feature for? thanks,

Hi @gagoar, thanks for the response. Rebased the branch onto the main.
If I need to do sth else I am happy to help!

gagoar commented:
> > Hey @BlackDark I've fixed some of the workflows, to make them run correctly when there's a fork, could you rebase with main? thanks!
> > Also please add a description of the use case you are trying to get this feature for? thanks,
> 
> Hi @gagoar, thanks for the response. Rebased the branch onto the main. If I need to do sth else I am happy to help!

yea, it seems like sadly the DeepSource is failing because the error on the try-catch doesn't have a type. are you familiar with TS to include it in your review? I can wip up a PR fixing that first as well. 


Also please add a description to the PR when you get a chance. Is very useful to know the use-case you are optimizing for. 

Thanks again.

gagoar commented:
The error:
![image](https://user-images.githubusercontent.com/226559/148742881-e97b5075-a753-4613-a8f6-59850af18479.png)

Can be solved with:

```ts
setFailed(`something went very wrong ${JSON.stringify((e as Error).message)}`);
```

gagoar commented:
Hey @BlackDark I've fixed the issue you see here coming from `DeepSource` if you please can *rebase your branch* and add a description to the PR* we will be good to go! 

I can help with  the action that uses this package as well so we can release things in tandem.
